### PR TITLE
[ASV-1153] downloads metrics

### PR DESCRIPTION
**Pull request by JDSM01**
**What does this PR do?**

 Adds the missing install complete event to facebook and flurry. Also adds the error event due to root that was missing since april 4th.
 
**Database changed?**

No

**Where should the reviewer start?**

-InstallAnalytics.java

**How should this be manually tested?**

Do a download in a non-editorial app (app view and apps) and check if the application install was sent to facebook and flurry and if the INSTALL event was sent to the BI.
 
**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/ASV-1153

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) 

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Partners build
- [ ] Unit tests pass
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/download/FileDownloadTask.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ protected void paused(BaseDownloadTask baseDownloadTask, long soFarBytes, long t
       Logger.getInstance()
           .d(TAG, " Download error");
       fileDownloadTaskStatus =
-          new FileDownloadTaskStatus(AppDownloadStatus.AppDownloadState.ERROR, md5);
+          new FileDownloadTaskStatus(AppDownloadStatus.AppDownloadState.ERROR, md5, null);
```
Comment (phillipsgoncalves) [inline app/src/main/java/cm/aptoide/pt/download/FileDownloadTask.java]: With this, errors while trying to calculate the md5 will not be logged.

File: app/src/main/java/cm/aptoide/pt/download/Md5DownloadComparisionException.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package cm.aptoide.pt.download;
+
+import cm.aptoide.pt.utils.BaseException;
+
+class Md5DownloadComparisionException extends BaseException {
```
Comment (phillipsgoncalves) [inline app/src/main/java/cm/aptoide/pt/download/Md5DownloadComparisionException.java]: Typo: Comparison not Comparision

**Comment by phillipsgoncalves** [comment]
Tested all events.
Problems:
Downgrade from editors Choice Application install event is being sent as Application install. This might be related with the ifs to check if it is editors choice or notification.

On Apps section in BN, when i do an update, it is not sending the download complete event to facebook and flurry, only sending the download event to BI. The Application install here is working fine.

**Comment by analaragomes** [comment]
https://ws75.aptoide.com/api/7/user/addEvent/name=INSTALL/action=INSTALL/context=APPVIEW
previous_context is wrong. In download event the previous_context is correct

**Comment by analaragomes** [comment]
For editors choice downloads:
Editors_Choice_Clicks => OK
Editors Choice_Download_Complete => OK
Download Complete => OK
Editors_Choice_Application_Install => OK
But Application Install is not sent.

**Comment by analaragomes** [comment]
https://ws75.aptoide.com/api/7/user/addEvent/name=DOWNLOAD/action=CLICK/context=Downloads => this context does not exist
https://ws75.aptoide.com/api/7/user/addEvent/name=INSTALL/action=CLICK/context=Downloads => this context does not exist

**Pull request merged**